dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -1475,7 +1475,9 @@ void NO_INLINE Aggregator::mergeDataImpl(
     Table & table_src,
     Arena * arena) const
 {
-    for (auto it = table_src.begin(); it != table_src.end(); ++it)
+    decltype(table_src.end()) end = table_src.end();
+
+    for (auto it = table_src.begin(); it != end; ++it)
```

Comment:
You can write
`for (auto it = table_src.begin(), end = table_src.end(); it != end; ++it)`
for more convenience.

Reply:
Fixed, thank you.